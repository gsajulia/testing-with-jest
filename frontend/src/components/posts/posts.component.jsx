/* Styles */
import './posts.styles.css';

/* Components */
import { PostCard } from '../post-card/post-card.component';

export const Posts = ({ posts = [] }) => (
  <div className="posts">
    {posts.length !== 0 ? posts.map(post => (
      <PostCard
        key={post.id}
        title={post.title}
        body={post.body}
        id={post.id}
        cover={post.cover}
      />
    )) : <p>There are no posts =(</p>}
  </div>
);