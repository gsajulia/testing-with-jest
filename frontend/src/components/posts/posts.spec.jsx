import { render, screen } from '@testing-library/react';
import { Posts } from './posts.component'

const props = {
  posts: [
    {
      id: 1,
      title: '1 - Harry Potter',
      body: 'This character interpeted by Daniel Radcliffe is a human of Gryffindor house studant at Hogwarts and born at 31-07-1980 he is half-blood, has green eyes and black hair, his patronus is a stag and his wand core is phoenix feather and is made with holly wood.',
      cover: 'img/img1.png',
    },
    {
      id: 2,
      title: '2 - Hermione Granger',
      body: 'This character interpeted by Emma Watson is a human of Gryffindor house studant at Hogwarts and born at 19-09-1979 she is muggleborn, has brown eyes and brown hair, her patronus is a otter and her wand core is dragon heartstring and is made with vine wood.',
      cover: 'img/img2.png',
    },
    {
      id: 3,
      title: '3 - Ron Weasley',
      body: 'This character interpeted by Rupert Grint is a human of Gryffindor house studant at Hogwarts and born at 01-03-1980 he is pure-blood, has blue eyes and red hair, his patronus is a Jack Russell terrier and his wand core is unicorn tail-hair and is made with willow wood.',
      cover: 'img/img3.png',
    },
  ]};

describe('<Posts />', () => {
    it('should render posts', () => {
      render(<Posts {...props} />);
  
      expect(screen.getAllByRole('heading', { name: props.posts.title}))
        .toHaveLength(3);
      expect(screen.getAllByRole('img', { name: / - /i }))
        .toHaveLength(3);
      expect(screen.getAllByText(/Gryffindor house/i))
        .toHaveLength(3);
        expect(screen.getByRole('img', { name: /3 - Ron Weasley/i }))
        .toHaveAttribute('src', 'img/img3.png');
    });
  
    it('should not render posts', () => {
      render(<Posts />);
      expect(screen.queryByRole('heading', { name: / - /i }))
        .not.toBeInTheDocument();
      expect(screen.getByText("There are no posts =("))
        .toBeInTheDocument();
    });
    
    it('should match snapshot', () => {
      const {container} = render(<Posts {...props} />);
      expect(container.firstChild).toMatchSnapshot();
    });
});