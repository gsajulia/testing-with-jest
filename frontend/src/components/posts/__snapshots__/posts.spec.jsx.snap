// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="1 - Harry Potter"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h1>
        <b>
          1
           - 
        </b>
        1 - Harry Potter
      </h1>
      <p>
        This character interpeted by Daniel Radcliffe is a human of Gryffindor house studant at Hogwarts and born at 31-07-1980 he is half-blood, has green eyes and black hair, his patronus is a stag and his wand core is phoenix feather and is made with holly wood.
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="2 - Hermione Granger"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        <b>
          2
           - 
        </b>
        2 - Hermione Granger
      </h1>
      <p>
        This character interpeted by Emma Watson is a human of Gryffindor house studant at Hogwarts and born at 19-09-1979 she is muggleborn, has brown eyes and brown hair, her patronus is a otter and her wand core is dragon heartstring and is made with vine wood.
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="3 - Ron Weasley"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        <b>
          3
           - 
        </b>
        3 - Ron Weasley
      </h1>
      <p>
        This character interpeted by Rupert Grint is a human of Gryffindor house studant at Hogwarts and born at 01-03-1980 he is pure-blood, has blue eyes and red hair, his patronus is a Jack Russell terrier and his wand core is unicorn tail-hair and is made with willow wood.
      </p>
    </div>
  </div>
</div>
`;
